### PR TITLE
Unknown edns 3.4.6

### PR DESCRIPTION
**Pull request by cmouse**
Backport of #2691 to 3.4.6. Tested against ISC EDNS compliance tester.

### CODE REVIEW

**Comment by cmouse** [comment]
https://ednscomp.isc.org/ednscomp/be7ab06b73 compliance test result

File: pdns/dnspacket.hh [hunk 1]
```diff
@@ -167,6 +173,9 @@ private:
   string d_rawpacket; // this is where everything lives 4
   int d_maxreplylen;
   string d_ednsping;
+  uint8_t d_ednsversion;
+  uint8_t d_ednsrcode;
```
Comment (Habbie) [inline pdns/dnspacket.hh]: EDNS rcode could theoretically be 12 bits, maybe note this in a comment to reduce surprises for whomever has to deal with that 30 years from now?

**Comment by cmouse** [comment]
Ah, you're right. I'll add this as well. I'll remake this pr after the main one has been rectified.

**Comment by cmouse** [comment]
https://ednscomp.isc.org/ednscomp/65cbd3f9da compliance check result after sync.

**Pull request merged**